### PR TITLE
Giza bug fixed no urls

### PR DESCRIPTION
**Pull request by dobertRowneySr**
reference #2726

### CODE REVIEW

File: runtime-modules/content/src/lib.rs [hunk 1]
```diff
@@ -1589,16 +1504,17 @@ decl_event!(
...
             ContentActor,
             PersonId,
-            NewAssets,
-            PersonUpdateParameters<NewAssets>,
+            StorageAssets,
+            PersonUpdateParameters<StorageAssets>,
         ),
         PersonDeleted(ContentActor, PersonId),
         ChannelDeleted(ContentActor, ChannelId),
+        //        VideoAssetsDeleted(ContentActor, VideoId, BTreeSet<AssetsToBeRemoved>),
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs]: Please, delete commented code.

File: runtime-modules/content/src/lib.rs [hunk 2]
```diff
@@ -1469,6 +1353,37 @@ impl<T: Trait> Module<T> {
...
+        Ok(())
+    }
+
+    fn remove_assets_from_storage(
+        assets: &BTreeSet<<T as storage::Trait>::DataObjectId>,
+        channel_id: &T::ChannelId,
+        prize_source_account: &T::AccountId,
+    ) -> DispatchResult {
+        // remove assets if any
+        if assets.len() > 0 {
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs]:
```suggestion
        if !assets.is_empty() {
```

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks OK.

There is a couple of minor issues. Also, please introduce some type alias for `<T as storage::Trait>::DataObjectId` and replace all occurrences.

File: runtime-modules/content/src/lib.rs [hunk 3]
```diff
@@ -706,9 +676,8 @@ decl_module! {
             actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
             channel_id: T::ChannelId,
             params: ChannelUpdateParameters<T>,
+            assets: BTreeSet<DataObjectId<T>>,
```
Comment (Lezek123) [inline runtime-modules/content/src/lib.rs]: I suggest renaming it to `assets_to_remove` for more clarity. Perhaps also moving it to `ChannelUpdateParameters` may be a good idea? (and analogically with video)

Comment (ignazio-bovo) [reply runtime-modules/content/src/lib.rs:679]: Ok I believe understand your thinking @Lezek123, ("Why do I need to specify the assets when I have to delete the whole bag regardless?") The problem lies in the fact that first the user is required to empty the channel bag by specifying objects ids. And there are not ways (at least that I am aware of) to retrieve such ids from the on-chain state

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

File: runtime-modules/content/src/lib.rs [hunk 4]
```diff
@@ -727,7 +728,7 @@ decl_module! {
             origin,
             actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
             channel_id: T::ChannelId,
-            assets_to_remove: BTreeSet<DataObjectId<T>>,
+            num_objects_to_delete: u64,
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs:731]: Why do we need this number? What we're trying to achieve using this check `storage::Bags::<T>::get(&bag_id).objects_number == num_objects_to_delete` ?

Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs:731]: https://github.com/Joystream/joystream/issues/2726#issuecomment-933485454

**Review by shamil-gadelshin — APPROVED** [review]
Still LGTM

**Review by Lezek123 — APPROVED** [review]
LGTM

**Pull request merged**